startup/30-basler.py

```diff
@@ -0,0 +1,118 @@
+        self.camera_serial_no = self.camera_object.GetDeviceInfo().GetSerialNumber()
+        self.width = self.camera_object.Width()
+        self.height = self.camera_object.Height()
+        self.pixel_level_min = self.camera_object.PixelDynamicRangeMin()
+        self.pixel_level_max = self.camera_object.PixelDynamicRangeMax()
+        self.active_format = self.camera_object.PixelFormat.GetValue()
+        self.formats_supported = self.camera_object.PixelFormat.Symbolics
+        self.payload_size = self.camera_object.PayloadSize()
+        self.grab_timeout = 5000 # ms
+
+        self.camera_object.Close()
+
+    def grab_images(self, num=10):
+
+        self.camera_object.StartGrabbingMax(num)
+        counter = itertools.count()
+        first_image = np.zeros((self.height, self.width))
+        while self.camera_object.IsGrabbing():
+            grab_result = self.camera_object.RetrieveResult(self.grab_timeout, pylon.TimeoutHandling_ThrowException)
+            if grab_result.GrabSucceeded():
+                image = grab_result.Array
+                current_frame = next(counter)
+                if current_frame == 1:
+                    first_image = np.copy(image)
+                print(
+                    f"{datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S.%f')} "
+                    f"{current_frame:04d} {image.shape = } {image.mean() = } "
+                    f"{image[0, 0] = } {image[-1, -1] = }"
+                )
+
+            grab_result.Release()
+
+        self.camera_object.StopGrabbing()
+
+        return image
+
+    def trigger(self, verbose=True):
+
+        super().trigger()
+
+        self.camera_object.Open()
+
+        trigger_mode = "Off"
+
+        if verbose:
+            print("User-defined camera name    :", self.user_defined_name)
+            print("Camera model                :", self.camera_model)
+            print("Camera serial number        :", self.camera_serial_no)
+            print("Image width  (X, ncols)     :", self.width, "pixels")
+            print("Image height (Y, nrows)     :", self.height, "pixels")
+            print("Pixel format                :", self.active_format)
+            print("Camera min. pixel level     :", self.pixel_level_min)
+            print("Camera max. pixel level     :", self.pixel_level_max)
+            print("Grab timeout                :", self.grab_timeout, "ms")
+            print("Trigger mode                :", trigger_mode)
+            print("GigE transport payload size : " + "{:,}".format(self.payload_size) + " bytes")
+            print("\nCamera supported pixel formats:\n", self.formats_supported)
+
+        self.camera_object.TriggerMode.SetValue(trigger_mode)
+        desired_pixel_format = "Mono8"
+        self.camera_object.PixelFormat.SetValue(desired_pixel_format)
+
+        image = self.grab_images(num=10)
+        self.update_components(image)
+        self.camera_object.Close()
```

Comment:
This `...Close()` call will better happen in the `unstage` method. That method should also revert the parameters to their original values.